components/WeeklyOverview/index.js

```diff
@@ -59,30 +104,41 @@ export default function WeeklyOverview() {
           <Weekday key={weekday}>
             <WeekdayContent>
               <WeekdayName>{weekday}</WeekdayName>
-              <MedicationSelect
-                value={selectedMedicationsByWeekday[weekday].medicationName}
+              <SupplementSelect
+                value={selectedSupplementsByWeekday[weekday].medicationName}
                 onChange={(event) => {
-                  const selectedMedication = dummyMedications.find(
-                    (med) => med.medicationName === event.target.value
+                  const selectedSupplement = dummyMedications.find(
+                    (supplement) => supplement.medicationName === event.target.value
                   );
-                  setSelectedMedicationsByWeekday((prevSelectedMeds) => ({
-                    ...prevSelectedMeds,
-                    [weekday]: selectedMedication,
+                  setSelectedSupplementsByWeekday((prevSelectedSupps) => ({
+                    ...prevSelectedSupps,
+                    [weekday]: selectedSupplement,
                   }));
                 }}
               >
-                {dummyMedications.map((med) => (
-                  <option key={med.id} value={med.medicationName}>
-                    {med.medicationName}
+                {dummyMedications.map((supplement) => (
+                  <option key={supplement.key} value={supplement.medicationName}>
+                    {supplement.medicationName}
                   </option>
                 ))}
-              </MedicationSelect>
+              </SupplementSelect>
               <AddButton onClick={() => handleAddSupplement(weekday)}>Add</AddButton>
             </WeekdayContent>
             {supplementsByWeekday[weekday].map((supplement) => (
-              <SupplementWrapper key={supplement.id}>
+              <SupplementWrapper key={`${supplement.medicationName}-${supplement.dosage}`}>
```

Comment:
This composite key might still not be unique when you enter the same med twice. For example, if I enter "Aspirin" and "100mg" twice, then this key is no longer unique.